### PR TITLE
merge from downstream

### PR DESCRIPTION
**Pull request by yagee-de**
clean up formatting and order to make a minimal change set

### CODE REVIEW

**Comment by kkrebs** [comment]
description from old pr

Hello,

as an user I really like your OOP-approach with METS/MODS.
Now I'd like to contribute some additional functionality regarding metsHdr-section that I do require for working with the mets-model-library in the context of mass-digitalization at the ULB. It consists of several extensions to the metsHdr-section that were lacking. This forced me to implement some rather hacky stuff to get around this limitation but I guess the proper way is to have these functionalities included in your nice library itself.

I needed to alter the existing metsHdr date handling, since I did get it to work properly at de-serialization with the previous Instant-types. Maybe you can come up with a better idea to handle this I'm looking forward!

Unfortunately it can't be merged out of the box since it is based on the latest published version (tag 2.0), which is not the one the master is even with.

**Comment by M3ssman** [comment]
@yagee-de @kkrebs , do you see any chances to include this into master in a reasonable amount of time?

File: src/main/java/org/mycore/mets/model/header/MetsHdr.java [hunk 1]
```diff
@@ -19,16 +22,25 @@
  */
 public class MetsHdr implements IMetsElement {
 
+    public static final DateTimeFormatter DT_FORMATTER = new DateTimeFormatterBuilder()
+        .appendPattern("uuuu-MM-dd")
+        .appendLiteral('T')
+        .appendPattern("HH:mm:ss")
+        .toFormatter();
```
Comment (yagee-de) [inline src/main/java/org/mycore/mets/model/header/MetsHdr.java]: CREATEDATE and LASTMODDATE are of type xsd:dateTime this Formatter is not sufficient

File: src/main/java/org/mycore/mets/model/header/MetsHdr.java [hunk 2]
```diff
@@ -19,16 +22,25 @@
...
     private String id;
 
     private String admId;
 
-    private Instant createDate;
+    private LocalDateTime createDate;
 
-    private Instant lastModDate;
+    private LocalDateTime lastModDate;
```
Comment (yagee-de) [inline src/main/java/org/mycore/mets/model/header/MetsHdr.java]: Local fields should stay as Instant

File: src/main/java/org/mycore/mets/model/header/MetsHdr.java [hunk 3]
```diff
@@ -37,10 +49,13 @@ public MetsHdr() {
         this.id = null;
         this.admId = null;
         this.recordStatus = null;
-        this.setCreateDate(Instant.now());
-        this.setLastModDate(Instant.now());
+        this.setCreateDate(LocalDateTime.now(ZoneId.of("Europe/Paris")));
+        this.setLastModDate(LocalDateTime.now(ZoneId.of("Europe/Paris")));
```
Comment (yagee-de) [inline src/main/java/org/mycore/mets/model/header/MetsHdr.java]: This assumption is wrong most of the times

File: src/main/java/org/mycore/mets/model/header/MetsHdr.java [hunk 4]
```diff
@@ -144,8 +166,8 @@ public Instant getLastModDate() {
      * 
      * @param lastModDate new modified date
      */
-    public void setLastModDate(Instant lastModDate) {
-        this.lastModDate = Instant.ofEpochMilli(lastModDate.toEpochMilli()).with(ChronoField.MILLI_OF_SECOND, 0);
+    public void setLastModDate(LocalDateTime lastModDate) {
+        this.lastModDate = lastModDate;
```
Comment (yagee-de) [inline src/main/java/org/mycore/mets/model/header/MetsHdr.java]: This should stay as Instant

**Comment by yagee-de** [comment]
I am ready for review: reformatted code and use forked and simplified code from mycore to store xs:dataTime in metsHdr

**Review by kkrebs — APPROVED** [review]

**Pull request merged**